src/components/Viewport.ts

```diff
@@ -608,14 +608,26 @@ export default class Viewport {
 
   public updateBbox(): BoundingBox {
     const state = this.state;
+    const options = this.options;
+    const viewportElement = this.viewportElement;
+
     if (!state.cachedBbox) {
-      const bbox = this.viewportElement.getBoundingClientRect();
-      state.cachedBbox = {
-        x: bbox.left,
-        y: bbox.top,
-        width: bbox.width,
-        height: bbox.height,
-      };
+      if (options.useOffset) {
```

Comment:
@WoodNeck This code is almost same with Panels' code. Is there any solution to use it commonly.

Reply:
I'll refactor that, thanks.